### PR TITLE
Another attempt at fix for #1580 (and undo #1567)

### LINKED ISSUE #1580: Integration test failure in CI due to failure to connect to Docker engine
(see #1567 for some context)

Failure 

```
Step #4 - "cassandra-3.11": ERROR [main] 2022-01-19 19:06:27,600 DockerClientProviderStrategy.java:208 - Could not find a valid Docker environment. Please check configuration. Attempted configurations were:
Step #4 - "cassandra-3.11": ERROR [main] 2022-01-19 19:06:27,600 DockerClientProviderStrategy.java:210 -     UnixSocketClientProviderStrategy: failed with exception TimeoutException (Timeout waiting for result with exception). Root cause IOException (native connect() failed : Permission denied)
Step #4 - "cassandra-3.11": ERROR [main] 2022-01-19 19:06:27,600 DockerClientProviderStrategy.java:212 - As no valid configuration was found, execution cannot continue
Step #4 - "cassandra-3.11": [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 30.417 s <<< FAILURE! - in io.stargate.it.cql.JwtAuthTest
Step #4 - "cassandra-3.11": [ERROR] io.stargate.it.cql.JwtAuthTest  Time elapsed: 30.417 s  <<< ERROR!
Step #4 - "cassandra-3.11": java.lang.reflect.InvocationTargetException
Step #4 - "cassandra-3.11": 	at io.stargate.it.cql.JwtAuthTest.buildParameters(JwtAuthTest.java:53)
Step #4 - "cassandra-3.11": 
```

causes occasional failures of 3 integration tests that rely on `testcontainers`.

Would be good to fix this problem.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Tries to prevent 3 transient IT failures on CI, based on:

https://dhananjay4058.medium.com/solving-docker-permission-denied-while-trying-to-connect-to-the-docker-daemon-socket-2e53cccffbaa

**Which issue(s) this PR fixes**:
Fixes #1580 (I hope)

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: ci/test.sh [hunk 1]
```diff
@@ -13,8 +13,9 @@ chown -R ubuntu:ubuntu /cache/
 echo '127.0.1.11 internal-stargate.local' >> /etc/hosts
 echo '127.0.1.12 internal-stargate.local' >> /etc/hosts
 
-# restart docker
-service docker --full-restart
+# Ensure we can connect to Docker daemon to resolve issue #1580, see:
+# https://dhananjay4058.medium.com/solving-docker-permission-denied-while-trying-to-connect-to-the-docker-daemon-socket-2e53cccffbaa
+sudo usermod -a -G docker ubuntu
```
Comment (jeffreyscarpenter) [inline ci/test.sh:18]: `sudo` is not required here. at this point in the script, you are running as root. line 20 is where it switches to `ubuntu`

Comment (tatu-at-datastax) [reply ci/test.sh:18]: Ok. I thought I saw `sudo` earlier but it isn't actually used so my bad.

**Review by jeffreyscarpenter — APPROVED** [review]

File: ci/test.sh [hunk 2]
```diff
@@ -13,8 +13,9 @@ chown -R ubuntu:ubuntu /cache/
 echo '127.0.1.11 internal-stargate.local' >> /etc/hosts
 echo '127.0.1.12 internal-stargate.local' >> /etc/hosts
 
-# restart docker
-service docker --full-restart
+# Ensure we can connect to Docker daemon to resolve issue #1580, see:
+# https://dhananjay4058.medium.com/solving-docker-permission-denied-while-trying-to-connect-to-the-docker-daemon-socket-2e53cccffbaa
+usermod -a -G docker ubuntu
```
Comment (ivansenic) [inline ci/test.sh:18]: hmm adding a user to that group without restarting the docker service seems odd.. @tatu-at-datastax what was this motivated by? was my fix not working?

and bu the way this was already done in the image building: https://github.com/stargate/stargate/blob/5f6381da305a1f63725fea7df6f68df993eb70ab/ci/Dockerfile#L38-L42

so this is surely not the fix for the issue!

Comment (tatu-at-datastax) [reply ci/test.sh:18]: Ok, I can undo it then. The fix did not seem to have any effect; I had 2 out of 3 failures after fix went in.

Comment (tatu-at-datastax) [reply ci/test.sh:18]: Sorry about this, I was wondering if user/group permissions were set but did not actually check.

**Comment by tatu-at-datastax** [comment]
Note: this change is being reverted (see #1583)

**Pull request merged**